src/ray/raylet/scheduling/cluster_resource_manager.cc

```diff
@@ -22,7 +22,25 @@
 
 namespace ray {
 
-ClusterResourceManager::ClusterResourceManager() : nodes_{} {}
+ClusterResourceManager::ClusterResourceManager(instrumented_io_context &io_service)
+    : timer_(io_service) {
+  if (RayConfig::instance().use_ray_syncer()) {
```

Comment:
Q: this logic is technically independent from the syncer right? I.e., it will work although syncer is off (and the logic is not affected by CreateMessage and ReceiveMessage).

Reply:
Yes it is. I originally wanted to add it to ray syncer but feel the feature makes the ray syncer complicated and seems not fitting there. And seems this is the only place need this feature, so I move it here.